File: 6. Null pointer demo.cpp
// Topic: Null pointer demo
// Description: Initialize pointer to NULL and check before dereferencing
#include <iostream>
using namespace std;

int main()
{
    int* p = NULL;  // or nullptr in C++11
    if (p)
    {
        cout << *p << '\n';
    }
    else
    {
        cout << "Pointer is NULL, not dereferencing." << '\n';
    }
    return 0;
}

// Time Complexity: O(1).
// Space Complexity: O(1).

File: 6. Null pointer demo.py
"""
Topic: Null pointer demonstration in Python
Description: Use None and check before 'dereferencing' (access)
"""
def main():
    p = None
    if p is None:
        print('Pointer is None, not accessing.')
    else:
        print(p[0])

if __name__ == '__main__':
    main()
    # Time Complexity: O(1).
    # Space Complexity: O(1).
